File: web/models/ticket-view.ts
export type ConditionItem = {
  field_id: number | null
  field_key: string | null
  operator: string
  value: string | number | boolean | null | unknown[]
}

export type ColumnConfigItem = {
  field_id: number | null
  field_key: string | null
  visible: boolean
  sort_order: number
}

export type TicketView = {
  id: number
  tenant_id: number
  name: string
  is_enabled: boolean
  sort_order: number
  condition_logic: string
  conditions: ConditionItem[]
  group_field_id: number | null
  custom_columns_enabled: boolean
  columns_config: ColumnConfigItem[]
  created_at: string
  updated_at: string
}

export type CreateTicketViewPayload = {
  name: string
  condition_logic?: string
  conditions?: ConditionItem[]
  group_field_id?: number | null
  custom_columns_enabled?: boolean
  columns_config?: ColumnConfigItem[]
}

export type UpdateTicketViewPayload = Partial<CreateTicketViewPayload>

export type TicketViewSortItem = {
  id: number
  sort_order: number
}

export type TicketViewSortPayload = {
  items: TicketViewSortItem[]
}

export type TicketViewTogglePayload = {
  is_enabled: boolean
}

File: server/app/schemas/ticket_view.py
"""
Pydantic schemas for ticket view management
"""
from pydantic import BaseModel, ConfigDict, Field

from app.schemas.base import TimestampSchema, PaginatedResponse


class ConditionItem(BaseModel):
    field_id: int | None = None
    field_key: str | None = None
    operator: str
    value: str | int | float | list | dict | bool | None = None


class ColumnConfigItem(BaseModel):
    field_id: int | None = None
    field_key: str | None = None
    visible: bool = True
    sort_order: int = 0


class TicketViewBase(BaseModel):
    name: str = Field(..., max_length=128)
    condition_logic: str = Field(default="and", max_length=8)
    conditions: list[ConditionItem] = Field(default_factory=list)
    group_field_id: int | None = None
    custom_columns_enabled: bool = False
    columns_config: list[ColumnConfigItem] = Field(default_factory=list)


class TicketViewCreate(TicketViewBase):
    pass


class TicketViewUpdate(BaseModel):
    name: str | None = Field(None, max_length=128)
    condition_logic: str | None = Field(None, max_length=8)
    conditions: list[ConditionItem] | None = None
    group_field_id: int | None = None
    custom_columns_enabled: bool | None = None
    columns_config: list[ColumnConfigItem] | None = None


class TicketViewResponse(TicketViewBase, TimestampSchema):
    model_config = ConfigDict(from_attributes=True)
    id: int
    tenant_id: int
    is_enabled: bool
    sort_order: int


class TicketViewListResponse(PaginatedResponse):
    items: list[TicketViewResponse]


class TicketViewToggleRequest(BaseModel):
    is_enabled: bool


class TicketViewSortItem(BaseModel):
    id: int
    sort_order: int


class TicketViewSortRequest(BaseModel):
    items: list[TicketViewSortItem]
